src/fetch-data.js

```diff
@@ -0,0 +1,197 @@
+// @ts-check
```

Comment:
Please retain the JSdoc but loose the type checking, we don't need that for now and is not enforceable as we don't have a build pipeline for our project